storage/src/test/java/tech/pegasys/teku/storage/server/AbstractDatabaseTest.java

```diff
@@ -490,6 +496,53 @@ public void testShouldRecordFinalizedBlocksAndStatesInBatchUpdate()
     testShouldRecordFinalizedBlocksAndStates(StateStorageMode.ARCHIVE, true);
   }
 
+  @Test
+  public void shouldRecreateMerkleTreeFromDatabase() {
+    final DataStructureUtil dataStructureUtil = new DataStructureUtil();
+    DepositWithIndex firstDeposit = dataStructureUtil.randomDepositWithIndex(1L);
+    DepositWithIndex secondDeposit = dataStructureUtil.randomDepositWithIndex(2L);
+    eth1DepositChannel.addEth1Deposit(firstDeposit);
+    eth1DepositChannel.addEth1Deposit(secondDeposit);
+
+    assertThat(eth1DepositChannel.getDeposits().get(0)).isEqualTo(firstDeposit);
+    assertThat(eth1DepositChannel.getDeposits().get(1)).isEqualTo(secondDeposit);
+    MerkleTree m = database.getMerkleTree();
+    assertThat(m.getProof(firstDeposit.getData().hash_tree_root())).isNotEmpty();
+    assertThat(m.getProof(secondDeposit.getData().hash_tree_root())).isNotEmpty();
+  }
+
+  @Test
+  public void shouldRecreateDepositNavigableMap() {
+    final DataStructureUtil dataStructureUtil = new DataStructureUtil();
+    DepositWithIndex firstDeposit = dataStructureUtil.randomDepositWithIndex(1L);
+    DepositWithIndex secondDeposit = dataStructureUtil.randomDepositWithIndex(2L);
+    eth1DepositChannel.addEth1Deposit(firstDeposit);
+    eth1DepositChannel.addEth1Deposit(secondDeposit);
+
+    NavigableMap<UnsignedLong, DepositWithIndex> data = database.getDepositNavigableMap();
+    DepositWithIndex first = data.get(UnsignedLong.valueOf(1L));
+    assertThat(first.getData()).isEqualTo(firstDeposit.getData());
+    assertThat(first.getIndex()).isEqualTo(firstDeposit.getIndex());
+
+    DepositWithIndex second = data.get(UnsignedLong.valueOf(2L));
+    assertThat(second.getData()).isEqualTo(secondDeposit.getData());
+    assertThat(second.getIndex()).isEqualTo(secondDeposit.getIndex());
```

Comment:
Might be better to do:
```
assertThat(data)
        .containsExactly(
            // Proofs are not stored in the database
            entry(
                UnsignedLong.valueOf(1),
                new DepositWithIndex(firstDeposit.getData(), firstDeposit.getIndex())),
            entry(
                UnsignedLong.valueOf(2),
                new DepositWithIndex(secondDeposit.getData(), secondDeposit.getIndex())));
```

`DepositWithIndex` will need an equals and hashCode method as well (it should have one anyway given it's comparable).